packages/react-native/plugins/with-nx-metro.ts

```diff
@@ -1,24 +1,41 @@
-import { workspaceRoot } from '@nrwl/workspace/src/utils/app-root';
+import { workspaceLayout, workspaceRoot } from '@nrwl/devkit';
+import { join } from 'path';
+import { existsSync } from 'fs-extra';
+
 import { getResolveRequest } from './metro-resolver';
 
 interface WithNxOptions {
   debug?: boolean;
   extensions?: string[];
+  projectRoot?: string;
+  watchFolders?: string[];
 }
 
 export function withNxMetro(config: any, opts: WithNxOptions = {}) {
   const extensions = ['', 'ts', 'tsx', 'js', 'jsx', 'json'];
   if (opts.debug) process.env.NX_REACT_NATIVE_DEBUG = 'true';
   if (opts.extensions) extensions.push(...opts.extensions);
 
-  // Set the root to workspace root so we can resolve modules and assets
-  config.projectRoot = workspaceRoot;
+  config.projectRoot = opts.projectRoot || workspaceRoot;
 
   // Add support for paths specified by tsconfig
   config.resolver = {
     ...config.resolver,
     resolveRequest: getResolveRequest(extensions),
   };
 
+  let watchFolders = config.watchFolders || [];
+  watchFolders = watchFolders.concat([
+    join(workspaceRoot, 'node_modules'),
+    join(workspaceRoot, workspaceLayout().libsDir),
+    join(workspaceRoot, '.storybook'),
+  ]);
+  if (opts.watchFolders?.length) {
+    watchFolders = watchFolders.concat(opts.watchFolders);
+  }
+
+  watchFolders = watchFolders.filter((folder) => existsSync(folder));
```

Comment:
Is there an issue if we don't filters? Like if the folder doesn't exist, but is added afterwards?

Probably just an edge case that requires restart?

Reply:
yes. without filters, i got an error like `Error: ENOENT: no such file or directory, lstat '/Users/emilyxiong/Code/tmp/nx-react-native-68-1/.storybook'` if i create an app without storybook
it requires a restart when adding folders